### PR TITLE
feat(athena): graphical interface

### PR DESCRIPTION
**Pull request by raphaelvignes**
Enable Athena's Graphical Interface.
## Checklist

* [x] Unit tests for the changes exist
* [x] Tests pass on CI and coverage remains at 100%
* [x] Documentation reflects the changes where applicable

### CODE REVIEW

File: toucan_connectors/awsathena/awsathena_connector.py [hunk 1]
```diff
@@ -20,31 +23,25 @@ class AwsathenaDataSource(ToucanDataSource):
...
+    @classmethod
+    def get_form(cls, connector: 'AwsathenaConnector', current_config: dict[str, Any]):
+        return create_model(
+            'FormSchema',
+            database=strlist_to_enum('database', connector.available_dbs),
+            __base__=cls,
+        ).schema()
 
 
 class AwsathenaConnector(ToucanConnector):
```
Comment (jGundermann) [inline toucan_connectors/awsathena/awsathena_connector.py:47]: Maybe the Discoverable class is missing here ?

**Review by jGundermann — COMMENTED** [review]
LGTM

File: toucan_connectors/awsathena/awsathena_connector.py [hunk 2]
```diff
@@ -83,20 +84,60 @@ def _add_pagination_to_query(
             return f'SELECT * FROM ({cls._strip_trailing_semicolumn(query)}) LIMIT {limit};'
         return query
 
+    @cached_property
```
Comment (lukapeschke) [inline toucan_connectors/awsathena/awsathena_connector.py:87]: I'm not sure this is a good idea for long-running processes, maybe we could add a TTL to this cache ?

**Review by lukapeschke — APPROVED** [review]
Linter unhappy, but code LGTM :ok_hand:

**Pull request merged**